docs/core/diagnostics/app_is_leaking_memory_eventual_crash.md

```diff
@@ -0,0 +1,131 @@
+---
+title: Debugging a memory leak - .NET Core
+description: A tutorial walk-through, debugging a memory leak in .NET Core.
+author: sdmaclea
+ms.author: stmaclea
+ms.date: 10/14/2019
+---
+# Debugging a memory leak
+
+**This article applies to: ✓** .NET Core 3.0 SDK and later versions
+
+In this scenario, the endpoint will slowly start leaking memory and eventually will result in an out of memory exception. To diagnose this scenario, we need several key pieces of diagnostics data.
+
+## Preparing
+
+The tutorial uses:
+
+- [Sample debug target](sample-debug-target.md) to trigger the scenario.
+- [dotnet-trace](dotnet-trace.md) to list processes.
+- [dotnet-counters](dotnet-counters.md) to check managed memory usage.
+- [dotnet-dump](dotnet-dump.md) to collect and analyze a dump file.
+
+The tutorial assumes the sample and tools are ready to use and you are running on **Linux**.
+
+## Memory counters
+
+Before we dig into collecting diagnostics data to help us root cause this scenario, we need to convince ourselves that what we are actually seeing is a memory leak (memory growth). We can use the [dotnet-counters](dotnet-counters.md) tool to get at this information.
+
+Lets run the [Sample debug target](sample-debug-target.md).
+
+```dotnetcli
+dotnet run
+```
+
+Then find the process ID using:
+
+```dotnetcli
+dotnet-trace list-processes
+```
+
+Before causing the leak, lets check our managed memory counters:
```

Comment:
In here, you'll cause a leak on purpose with the app. But I think you should just state what the normal steps are.
For example:

```suggestion
Now, check the managed memory counters using the [dotnet-counters](dotnet-counters.md) global tool before the leak occurs. The `--refresh-interval` specifies the number of seconds between refreshes.
```